sources/packages/api/src/esdc-integration/e-cert-full-time-integration/e-cert-full-time-request.service.ts

```diff
@@ -0,0 +1,171 @@
+            nextSequenceNumber,
+          );
+          // Create the request filename with the file path for the e-Cert File.
+          const fileInfo =
+            await this.ecertIntegrationService.createRequestFileName(
+              entityManager,
+            );
+
+          // Creates the repository based on the entity manager that
+          // holds the transaction already created to manage the
+          // sequence number.
+          const disbursementScheduleRepo =
+            entityManager.getRepository(DisbursementSchedule);
+          this.disbursementScheduleService.updateRecordsInSentFile(
+            disbursementIds,
+            getUTCNow(),
+            disbursementScheduleRepo,
+          );
+
+          this.logger.log("Uploading content...");
+          await this.ecertIntegrationService.uploadContent(
+            fileContent,
+            fileInfo.filePath,
+          );
+
+          uploadResult = {
+            generatedFile: fileInfo.filePath,
+            uploadedRecords: disbursementRecords.length,
+          };
+        } catch (error) {
+          this.logger.error(
+            `Error while uploading content for e-Cert file: ${error}`,
+          );
+          throw error;
+        }
+      },
+    );
+    return uploadResult;
+  }
+
+  /**
+   * Create the e-Cert record with the information needed to generate the
+   * entire record to be sent to ESDC.
+   * @param disbursement disbursement that contains all information to
+   * generate the record.
+   * @returns e-Cert record.
+   */
+  private createECertRecord(disbursement: DisbursementSchedule): ECertRecord {
+    const now = new Date();
+    const addressInfo =
+      disbursement.application.student.contactInfo.addresses[0];
+    const fieldOfStudy = getFieldOfStudyFromCIPCode(
+      disbursement.application.offering.educationProgram.cipCode,
+    );
+    const awards = disbursement.disbursementValues.map(
+      (disbursementValue) =>
+        ({
+          valueType: disbursementValue.valueType,
+          valueCode: disbursementValue.valueCode,
+          valueAmount: round(disbursementValue.valueAmount).toString(),
+        } as Award),
+    );
+
+    return {
+      sin: disbursement.application.student.sin,
+      applicationNumber: disbursement.application.applicationNumber,
+      documentNumber: disbursement.documentNumber,
+      disbursementDate: disbursement.disbursementDate,
+      documentProducedDate: now,
+      negotiatedExpiryDate: disbursement.negotiatedExpiryDate,
+      schoolAmount:
+        disbursement.application.offering.tuitionRemittanceRequestedAmount,
+      educationalStartDate: disbursement.application.offering.studyStartDate,
+      educationalEndDate: disbursement.application.offering.studyEndDate,
+      federalInstitutionCode: disbursement.application.location.institutionCode,
+      weeksOfStudy: 1,
+      fieldOfStudy,
+      // TODO: get the value from offerings (to be defined/added).
```

Comment:
Thanks for adding the comment. Yes it is part of Program and Offering enhancements.